includes/wc-update-functions.php

```diff
@@ -1714,6 +1712,23 @@ function wc_update_340_states() {
 		),
 	);
 
+	update_option( 'woocommerce_update_340_states', $country_states );
```

Comment:
Any reason why you went with an option and not a transient here?

Reply:
So it persists reliably until the routine is complete.

Reply:
OK, but if for some reason an update times out on a single state update will it not go into an infinite loop again? No big deal and probably highly unlikely, just thought it could be an edge case.

Reply:
@kloon Yes it would. We cannot break it down more but we could update before the SQL?